doc/source/ray-overview/index.rst

```diff
@@ -0,0 +1,160 @@
+.. _gentle-intro:
+
+============================
+A Gentle Introduction to Ray
+============================
+
+.. include:: basics.rst
+
+This tutorial will provide a whirlwind tour of all that Ray has to offer.
+
+First, install Ray with: ``pip install ray``, now we can execute some Python in parallel.
+
+Parallelizing Python Functions with Ray Tasks
+=============================================
+
+First, import ray and ``init`` the Ray service.
+Then decorate your function with ``@ray.remote`` to declare that you want to run this function 
+remotely. Lastly, call that function with the ``.remote`` method call, this yields futures that you can then 
+fetch with ``ray.get``.
+
+.. code-block:: python
+
+    import ray
+    ray.init()
+
+    @ray.remote
+    def f(x):
+        return x * x
+
+    futures = [f.remote(i) for i in range(4)]
+    print(ray.get(futures)) # [0, 1, 4, 9]
+
+In the above code block we defined some Ray Tasks. While these are great for stateless operations sometimes you
+must maintain the state of your application. You can do that with Ray Actors.
+
+Parallelizing Python Functions with Ray Actors
+==============================================
+
+Ray provides Actors to allow you to parallelize an instance of a class in Python.
+When you instantiate a class that is a Ray Actor, Ray will automatically distribute
+that actor onto the Ray cluster and manage that state on your behalf. All you as a 
+user need to do is call the class using the ``remote`` method call.
+
+.. code-block:: python
+
+    import ray
+    ray.init() # only call this one
+ 
+    @ray.remote
+    class Counter(object):
+        def __init__(self):
+            self.n = 0
+
+        def increment(self):
+            self.n += 1
+
+        def read(self):
+            return self.n
+
+    counters = [Counter.remote() for i in range(4)]
+    [c.increment.remote() for c in counters]
+    futures = [c.read.remote() for c in counters]
+    print(ray.get(futures)) # [1, 1, 1, 1]
+
+
+An Overview of the Ray Libraries
+================================
+
+Ray has a vast ecosystem of libraries and frameworks built on top of it. The main ones being:
+
+- :ref:`tune-index`
+- :ref:`rllib-index`
+- :ref:`sgd-index`
+- :ref:`rayserve`
+
+
+Tune Quick Start
```

Comment:
Should these quick starts be factored out and used in the same place in the respective library docs and here?

Reply:
Yes, would like to do this in a follow up PR.